File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/models/explorer/UploadFile.java
package app.editors.manager.mvp.models.explorer;

import android.net.Uri;
import android.os.Parcel;
import android.os.Parcelable;

import java.util.Objects;

import app.editors.manager.mvp.models.base.Entity;
import app.editors.manager.ui.adapters.holders.factory.TypeFactory;

public class UploadFile implements Entity, Parcelable {

    private String id;
    private String folderId;
    private String name;
    private Uri uri;
    private String size;
    private int progress;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getFolderId() {
        return folderId;
    }

    public void setFolderId(String folderId) {
        this.folderId = folderId;
    }

    public Uri getUri() {
        return uri;
    }

    public void setUri(Uri uri) {
        this.uri = uri;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public int getProgress() {
        return progress;
    }

    public void setProgress(int progress) {
        this.progress = progress;
    }

    @Override
    public int getType(TypeFactory factory) {
        return factory.type(this);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        UploadFile that = (UploadFile) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(folderId, that.folderId) &&
                Objects.equals(name, that.name) &&
                Objects.equals(uri, that.uri) &&
                Objects.equals(size, that.size);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, folderId, name, uri, size);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.id);
        dest.writeString(this.folderId);
        dest.writeString(this.name);
        dest.writeParcelable(this.uri, flags);
        dest.writeString(this.size);
        dest.writeInt(this.progress);
    }

    public UploadFile() {
    }

    protected UploadFile(Parcel in) {
        this.id = in.readString();
        this.folderId = in.readString();
        this.name = in.readString();
        this.uri = in.readParcelable(Uri.class.getClassLoader());
        this.size = in.readString();
        this.progress = in.readInt();
    }

    public static final Parcelable.Creator<UploadFile> CREATOR = new Parcelable.Creator<UploadFile>() {
        @Override
        public UploadFile createFromParcel(Parcel source) {
            return new UploadFile(source);
        }

        @Override
        public UploadFile[] newArray(int size) {
            return new UploadFile[size];
        }
    };
}

File: app_manager/core/src/main/java/app/documents/core/network/manager/models/explorer/UploadFile.kt
package app.documents.core.network.manager.models.explorer

import android.net.Uri
import android.os.Parcel
import android.os.Parcelable
import app.documents.core.network.manager.models.base.Entity
import java.util.*

class UploadFile() : Entity, Parcelable {
    var id: String? = null
    var folderId: String? = null
    var name: String? = null
    var uri: Uri? = null
    var size: String? = null
    var progress = 0

    constructor(parcel: Parcel) : this() {
        id = parcel.readString()
        folderId = parcel.readString()
        name = parcel.readString()
        uri = parcel.readParcelable(Uri::class.java.classLoader)
        size = parcel.readString()
        progress = parcel.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as UploadFile
        return id == that.id &&
                folderId == that.folderId &&
                name == that.name &&
                uri == that.uri &&
                size == that.size
    }

    override fun hashCode(): Int {
        return Objects.hash(id, folderId, name, uri, size)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(id)
        parcel.writeString(folderId)
        parcel.writeString(name)
        parcel.writeParcelable(uri, flags)
        parcel.writeString(size)
        parcel.writeInt(progress)
    }

    companion object CREATOR : Parcelable.Creator<UploadFile> {
        override fun createFromParcel(parcel: Parcel): UploadFile {
            return UploadFile(parcel)
        }

        override fun newArray(size: Int): Array<UploadFile?> {
            return arrayOfNulls(size)
        }
    }

}